core/include/traccc/fitting/kalman_filter/gain_matrix_smoother.hpp

```diff
@@ -102,6 +104,14 @@ struct gain_matrix_smoother {
         cur_state.smoothed().set_vector(smt_vec);
         cur_state.smoothed().set_covariance(smt_cov);
 
+        // Return false if track is parallel to z-axis or phi is not finite
+        const scalar theta = cur_state.smoothed().theta();
+        if (theta <= 0.f || theta >= constant<traccc::scalar>::pi ||
+            !std::isfinite(cur_state.smoothed().phi()) ||
+            std::abs(cur_state.smoothed().qop()) == 0.f) {
```

Comment:
When do you hit the condition of `qop = 0`? I observed this from the buggy behavior of smoother. Once I fix the bug, this may not be observed

Reply:
Ah, it is very possible that your fix will solve this problem, too!

Reply:
Yeah but if you are having an issue from `qop=0` right know, I am onboard with it